### PR TITLE
Upgrade Apache commons compress to 1.25

### LINKED ISSUE #27604: Gradle only partly extracts zip files with extra field that is larger than 8024 bytes
### Current Behavior

When using the `Copy` task to extract a zip file with an entry that has an extra field larger than 8024 bytes that entry will be the last entry extracted.

### Expected Behavior

The `Copy` task should extract the whole zip file even when an entry has an extra field larger than 8024 bytes.

### Context (optional)

_No response_

### Steps to Reproduce

https://github.com/coehlrich/gradle-bug-extract-zip

Both zip files have files:
- `1.txt` with an empty extra field
- `2.txt` with an extra field where
  - in `8024.zip` is 8024 bytes in length
  - in `8025.zip` is 8025 bytes in length
- `3.txt` with an extra field that is 8024 bytes in length

Notice how when the `extract8024` task is run all 3 files are extracted while running `extract8025` only the first 2 files are extracted.

### Gradle version

8.5

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

### PR DESCRIPTION
**Pull request by ov7a**
Fixes #27604

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by cobexer — CHANGES_REQUESTED** [review]
This may need release notes /upgrade guide because some of the changes in commons-compress may change the produced files:

https://issues.apache.org/jira/browse/COMPRESS-612
* https://github.com/apache/commons-compress/pull/254

https://issues.apache.org/jira/browse/COMPRESS-613
* https://github.com/apache/commons-compress/pull/345

https://issues.apache.org/jira/browse/COMPRESS-638
* https://github.com/apache/commons-compress/commit/770ea82a132282fb0edb186fe6db646a1a0b7a35

possibly more

**Comment by ov7a** [comment]
@cobexer Yep. already working on that, sorry for miscommunication

**Review by lkasso — APPROVED** [review]
Docs LGTM

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -40,10 +40,16 @@ The previous step will help you identify potential problems by issuing deprecati
 
 === Potential breaking changes
 
+==== Upgrade to Apache Commons Compress 1.25.0
+
+Apache Commons Compress has been updated from 1.21 to https://commons.apache.org/proper/commons-compress/changes-report.html#a1.25.0[1.25.0].
```
Comment (cobexer) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:45]: 💅 might include that the previous version used to be 1.21

Comment (cobexer) [reply platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:45]: I... go get myself a coffee...

**Review by cobexer — APPROVED** [review]

**Comment by ov7a** [comment]
@bot-gradle merge

**Comment by ov7a** [comment]
@bot-gradle APT

**Comment by ov7a** [comment]
@bot-gradle test APT

**Comment by ov7a** [comment]
@bot-gradle merge

**Pull request merged**